core/dbt/contracts/graph/unparsed.py

```diff
@@ -736,10 +738,54 @@ def normalize_date(d: Optional[datetime.date]) -> Optional[datetime.datetime]:
     return dt
 
 
+class UnitTestFormat(StrEnum):
+    CSV = "csv"
+    Dict = "dict"
+
+
+class UnitTestFixture:
+    @property
+    def format(self) -> UnitTestFormat:
+        return UnitTestFormat.Dict
+
+    @property
+    def rows(self) -> Union[str, List[Dict[str, Any]]]:
+        return []
+
+    def get_rows(self) -> List[Dict[str, Any]]:
+        if self.format == UnitTestFormat.Dict:
+            assert isinstance(self.rows, List)
+            return self.rows
+        elif self.format == UnitTestFormat.CSV:
+            assert isinstance(self.rows, str)
+            dummy_file = StringIO(self.rows)
+            reader = csv.DictReader(dummy_file)
+            rows = []
+            for row in reader:
+                rows.append(row)
+            return rows
+
+    @classmethod
+    def validate_fixture(cls, fixture, fixture_type, test_name):
+        if (fixture.format == UnitTestFormat.Dict and not isinstance(fixture.rows, list)) or (
+            fixture.format == UnitTestFormat.CSV and not isinstance(fixture.rows, str)
+        ):
+            raise ParsingError(
+                f"Unit test {test_name} has {fixture_type} rows which do not match format"
```

Comment:
Could make this message a tiny bit clearer if we also provided the format, e.g "do not match format: {fixture.format}"

Reply:
Done.